universal.c

```diff
@@ -433,6 +433,29 @@ XS(XS_UNIVERSAL_isa)
     }
 }
 
+XS(XS_UNIVERSAL_import_unimport); /* prototype to pass -Wmissing-prototypes */
+XS(XS_UNIVERSAL_import_unimport)
+{
+    dXSARGS;
+    dXSI32;
+
+    if (items > 1) {
+        char *class_pv= SvPV_nolen(ST(0));
+        if (strEQ(class_pv,"UNIVERSAL"))
+            Perl_croak(aTHX_ "UNIVERSAL does not export anything");
+        /* _charnames is special - ignore it for now as the code that
+         * depends on it has its own "no import" logic that produces better
+         * warnings than this does. */
+        if (strNE(class_pv,"_charnames"))
+            Perl_croak(aTHX_
+                "Attempt to call undefined %s with arguments via package %s"
+                " (Perhaps you forgot to load \"%s\"?)",
+                ix ? "unimport" : "import", class_pv, class_pv);
```

Comment:
This will not work correctly for non-ASCII package names. Use `SVf` and `SVfARG` as described in [`perlguts`](https://perldoc.perl.org/perlguts#Formatted-Printing-of-SVs).

Reply:
Fixed. Thanks. I wonder, should it actually use something that would make unprintables visible?

Reply:
FWIW, I have pushed a set of follow up patches adding a way to deal with unprintables.